io/src/main/scala/fs2/io/udp/AsynchronousSocketGroup.scala

```diff
@@ -0,0 +1,146 @@
+
+    override def register(channel: DatagramChannel): (AsynchronousSocketGroup.Attachment, SelectionKey) = {
+      channel.configureBlocking(false)
+      val attachment = new AsynchronousSocketGroup.Attachment()
+      val key = {
+        selectorLock.lock
+        try {
+          selector.wakeup
+          channel.register(selector, 0, attachment)
+        } finally {
+          selectorLock.unlock
+        }
+      }
+      (attachment, key)
+    }
+
+    override def enqueue(f: => Unit): Unit = {
+      pendingThunks.synchronized {
+        pendingThunks.+=(() => f)
+      }
+      selector.wakeup()
+      ()
+    }
+
+    private val doneNow = new AtomicBoolean(false)
+    private val selectorThread = new Thread(new Runnable {
+      def run = try {
+        val readBuffer = ByteBuffer.allocate(1 << 16)
+        while (!doneNow.get && !Thread.currentThread.isInterrupted) {
+          pendingThunks.synchronized { while (pendingThunks.nonEmpty) pendingThunks.dequeue()() }
+          selectorLock.lock
+          selectorLock.unlock
+          println(selector.select)
+          val selectedKeys = selector.selectedKeys.iterator
+          while (selectedKeys.hasNext) {
+            val key = selectedKeys.next
+            selectedKeys.remove
+            val channel = key.channel.asInstanceOf[DatagramChannel]
+            val attachment = key.attachment.asInstanceOf[Attachment]
+            if (key.isValid) {
+              if (key.isReadable) {
+                readBuffer.clear
+                val src = channel.receive(readBuffer)
+                if (src ne null) {
+                  attachment.dequeueReader match {
+                    case Some(reader) =>
+                      readBuffer.flip
+                      val bytes = Array.ofDim[Byte](readBuffer.remaining)
+                      readBuffer.get(bytes)
+                      readBuffer.clear
+                      reader(Right(new Packet(src, Chunk.bytes(bytes))))
+                      if (!attachment.hasReaders) key.interestOps(key.interestOps & ~SelectionKey.OP_READ)
+                    case None =>
+                      sys.error("key marked for read but no reader")
+                  }
+                }
+              }
+              if (key.isWritable) {
+                attachment.peekWriter match {
+                  case Some((p, cb)) =>
+                    val sent = channel.send(ByteBuffer.wrap(p.bytes.toArray), p.remote)
+                    if (sent > 0) {
+                      attachment.dequeueWriter
+                      cb(None)
+                      if (!attachment.hasWriters) key.interestOps(key.interestOps & ~SelectionKey.OP_WRITE)
+                    }
+                  case None =>
+                    sys.error("key marked for write but no writer")
+                }
+              }
+            }
+          }
+        }
+      } catch { case t: Throwable => t.printStackTrace }
```

Comment:
Will do. This was just some debugging I had left here. I'll make this match TCP or Strategy.